### DISCUSSION TITLE
How to get `pdf-parse` to work on Vercel deployments?

### DISCUSSION THREAD

**Discussion by PabloGracia** [vercel/community#5278] [Help]
### Summary

Using `pdf-parse` has some quicks in NextJS. The package will give an error when you try to use it:

```
Error: ENOENT: no such file or directory, open './test/data/05-versions-space.pdf'
```

According to the community ([such as here](https://stackoverflow.com/questions/76424198/why-do-pdf-parsing-libraries-pdf2json-and-pdf-parse-seem-to-not-work-with-next-j) or [here](https://gitlab.com/autokent/pdf-parse/-/issues/24)) the simplest solution seems to be adding `test/data/05-versions-space.pdf` to my project. It doesn't matter what pdf it is, just that it has that name. That together with other user's suggestion have made it work in my local project:
![image](https://github.com/vercel/community/assets/46579253/241a2bc8-ddde-4c6c-ae77-c31dcd2b76bc)

However, when I deploy the app to Vercel, the same app that works locally, doesn't work on Vercel:
![image](https://github.com/vercel/community/assets/46579253/c6264367-a42b-4b60-9194-a3a1b672c533)

I know this is first and foremost an issue with the package itself, but is there any way that I can make it work in Vercel the same that it works locally?

### Steps to Reproduce

Basically any new project that uses the `pdf-parse` [package](https://www.npmjs.com/package/pdf-parse) runs into this issue

**Comment by gerkim62** [comment]
facing the same issue, my deployment is working on render.com and localhost but not on vercel

**Comment by gerkim62** [comment]
# Here is how I fixed it

## Step 1:

```
npm i patch-package
```

## step 2:  

```
npm i pdf-parse
```

## step 3:

Go to this file : node\_modules\\pdf-parse\\index.js  

## step 4:

Change this line

```
let isDebugMode = !module.parent; (for me is line 6)  
```

to this one

```
let isDebugMode = false//!module.parent;
```

Just convert the isDebugMode to false. (I assume you understand what this will do if you read the code)


## step 5:  

run this command

```
npx patch-package pdf-parse
```

## step 6:  

go to the package.json and add this postinstall script  

```json
"scripts": {
//your other scripts  

"postinstall": "patch-package" 
}
```

## step 7:  

delete this folder  

.next\cache

## step 8:  
Go to your vercel deployment and redeploy without using the existing build cache (!important).
this ☝☝☝☝

Note:

if it does not work (not likely)

delete the package-lock.json and run  

```
npm install  
```

(am not sure if deleting package-lock.json is risky).

Let me know if it works for you

**Reply by kashishgadhiya** [reply]
not working

**Reply by gerkim62** [reply]
did it at least give u a different error message

**Reply by gerkim62** [reply]
https://stackoverflow.com/questions/76424198/why-do-pdf-parsing-libraries-pdf2json-and-pdf-parse-seem-to-not-work-with-next-j/77722836#77722836

please look at this SO question

**Reply by Suzzit6** [reply]
worked for me thank u so much !!

**Comment by diegocasmo** [comment]
`import * as parser from "pdf-parse/lib/pdf-parse.js";` works for me
